text/0000-source-files-mtime.md

```diff
@@ -0,0 +1,97 @@
+# Meta
+[meta]: #meta
+- Name: Application source files mtime
+- Start Date: 2025-03-28
+- Author(s): Infra-Red
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: "N/A"
+
+# Summary
+[summary]: #summary
+
+Currently, the modification time (mtime) of application source files within images created by Cloud Native Buildpacks is hard-coded to January 1, 1980. This behavior results in web caching issues because the `Last-Modified` HTTP header is always set to `Tue, 01 Jan 1980 00:00:01 GMT`. This RFC proposes adding options to either preserve the original modification time (mtime) of source files or set it based on the value provided in the `SOURCE_DATE_EPOCH` environment variable.
+
+# Definitions
+[definitions]: #definitions
+
+mtime - modified timestamp (mtime) indicates the last time the contents of a file were modified. For example, if new contents were added, deleted, or replaced in a file, the modified timestamp is changed.
+
+# Motivation
+[motivation]: #motivation
+
+### Why should we do this?
+
+This feature has been requested by the community (https://github.com/buildpacks/lifecycle/issues/1358), and the issue was demonstrated in the [matthewrobertson/bp-mtime-issue](https://github.com/matthewrobertson/bp-mtime-issue) repository.
+
+### What use cases does it support?
+
+This is important for the users who serve static content from their applications as they want to make sure that clients are always getting updated content.
+
+### What is the expected outcome?
+
+Lifecycle will be capable of copying the source files into the application image with original mtime preserved or set to a user-provided value.
```

Comment:
There is a lifecycle issue here: https://github.com/buildpacks/lifecycle/issues/1358

This [comment](https://github.com/buildpacks/lifecycle/issues/1358#issuecomment-2153060080) lists out some subtleties that would be good to address here, such as what should happen if buildpacks modify the source directory.